File: SDKs/xero.com/xero-payroll-au/2.9.4/typescript/src/sdk/models/shared/deductiontype.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum DeductionTypeDeductionCategoryEnum {
    None = "NONE",
    Unionfees = "UNIONFEES",
    Workplacegiving = "WORKPLACEGIVING"
}


export class DeductionType extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=AccountCode" })
  accountCode?: string;

  @SpeakeasyMetadata({ data: "json, name=CurrentRecord" })
  currentRecord?: boolean;

  @SpeakeasyMetadata({ data: "json, name=DeductionCategory" })
  deductionCategory?: DeductionTypeDeductionCategoryEnum;

  @SpeakeasyMetadata({ data: "json, name=DeductionTypeID" })
  deductionTypeID?: string;

  @SpeakeasyMetadata({ data: "json, name=IsExemptFromW1" })
  isExemptFromW1?: boolean;

  @SpeakeasyMetadata({ data: "json, name=Name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=ReducesSuper" })
  reducesSuper?: boolean;

  @SpeakeasyMetadata({ data: "json, name=ReducesTax" })
  reducesTax?: boolean;

  @SpeakeasyMetadata({ data: "json, name=UpdatedDateUTC" })
  updatedDateUTC?: string;
}


export class DeductionTypeInput extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=AccountCode" })
  accountCode?: string;

  @SpeakeasyMetadata({ data: "json, name=CurrentRecord" })
  currentRecord?: boolean;

  @SpeakeasyMetadata({ data: "json, name=DeductionCategory" })
  deductionCategory?: DeductionTypeDeductionCategoryEnum;

  @SpeakeasyMetadata({ data: "json, name=DeductionTypeID" })
  deductionTypeID?: string;

  @SpeakeasyMetadata({ data: "json, name=IsExemptFromW1" })
  isExemptFromW1?: boolean;

  @SpeakeasyMetadata({ data: "json, name=Name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=ReducesSuper" })
  reducesSuper?: boolean;

  @SpeakeasyMetadata({ data: "json, name=ReducesTax" })
  reducesTax?: boolean;
}

File: SDKs/xero.com/xero-payroll-au/2.9.4/python/src/sdk/models/shared/deductiontype.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils

class DeductionTypeDeductionCategoryEnum(str, Enum):
    NONE = "NONE"
    UNIONFEES = "UNIONFEES"
    WORKPLACEGIVING = "WORKPLACEGIVING"


@dataclass_json
@dataclasses.dataclass
class DeductionType:
    account_code: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('AccountCode') }})
    current_record: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('CurrentRecord') }})
    deduction_category: Optional[DeductionTypeDeductionCategoryEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('DeductionCategory') }})
    deduction_type_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('DeductionTypeID') }})
    is_exempt_from_w1: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('IsExemptFromW1') }})
    name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Name') }})
    reduces_super: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ReducesSuper') }})
    reduces_tax: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ReducesTax') }})
    updated_date_utc: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('UpdatedDateUTC') }})
    

@dataclass_json
@dataclasses.dataclass
class DeductionTypeInput:
    account_code: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('AccountCode') }})
    current_record: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('CurrentRecord') }})
    deduction_category: Optional[DeductionTypeDeductionCategoryEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('DeductionCategory') }})
    deduction_type_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('DeductionTypeID') }})
    is_exempt_from_w1: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('IsExemptFromW1') }})
    name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Name') }})
    reduces_super: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ReducesSuper') }})
    reduces_tax: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ReducesTax') }})
    
